File: web/src/api/scheduled_task.ts
// 定时任务配置
import apiClient from "@/api/client.ts";

export interface ScheduledTask {
    id: string;
    name: string;
    enabled: boolean;
    intervalDays: number;
    phoneNumber: string;
    content: string;
    createdAt?: number;
    lastRunAt?: number;
}

// 定时任务 API (RESTful)
// 获取所有定时任务
export const getScheduledTasks = () => {
    return apiClient.get<ScheduledTask[]>('/scheduled-tasks');
};

// 获取单个定时任务
export const getScheduledTask = (id: string) => {
    return apiClient.get<ScheduledTask>(`/scheduled-tasks/${id}`);
};

// 创建定时任务
export const createScheduledTask = (task: Omit<ScheduledTask, 'id' | 'createdAt' | 'lastRunAt'>) => {
    return apiClient.post<ScheduledTask>('/scheduled-tasks', task);
};

// 更新定时任务
export const updateScheduledTask = (id: string, task: Omit<ScheduledTask, 'id' | 'createdAt' | 'lastRunAt'>) => {
    return apiClient.put<ScheduledTask>(`/scheduled-tasks/${id}`, task);
};

// 删除定时任务
export const deleteScheduledTask = (id: string) => {
    return apiClient.delete<{ message: string }>(`/scheduled-tasks/${id}`);
};

File: internal/models/scheduled_task.go
package models

// ScheduledTask 定时任务
type ScheduledTask struct {
	ID           string `gorm:"primaryKey" json:"id"`                  // UUID
	Name         string `json:"name"`                                  // 任务名称
	Enabled      bool   `json:"enabled"`                               // 是否启用
	IntervalDays int    `json:"intervalDays"`                          // 执行间隔天数，例如 90 表示每90天执行一次
	PhoneNumber  string `json:"phoneNumber"`                           // 目标手机号
	Content      string `gorm:"type:text" json:"content"`              // 短信内容
	CreatedAt    int64  `json:"createdAt"`                             // 创建时间（时间戳毫秒）
	UpdatedAt    int64  `json:"updatedAt" gorm:"autoUpdateTime:milli"` // 更新时间（时间戳毫秒）
	LastRunAt    int64  `json:"lastRunAt"`                             // 上次执行时间（时间戳毫秒）
}

func (ScheduledTask) TableName() string {
	return "scheduled_tasks"
}
